src/d3a/models/area/redis_external_connection.py

```diff
@@ -11,6 +11,31 @@ def __init__(self, area):
         self.pubsub = self.redis_db.pubsub()
         self.sub_to_area_event()
         self.areas_to_register = []
+        self.areas_to_unregister = []
+
+    def market_cycle_event(self):
+        self.unregister_pending_areas()
+        self.register_new_areas()
+
+    def unregister_pending_areas(self):
+        if not self.areas_to_unregister:
+            return
+
+        for area in self.areas_to_unregister:
+            try:
+                area_object = next(child for child in self.area.children if child.name == area)
+                area_object.deactivate()
+                self.area.children.remove(area_object)
+            except Exception as e:
+                self.area.log.error(f"Unsubscribing of area {area} failed with error {str(e)}.")
+                self.publish(f"{self.area.slug}/unregister_participant/response",
+                             json.dumps({"response": "failed"}))
+            else:
```

Comment:
Shouldn't here be an `if` block first before this `else` block?

Reply:
This is not an if/else block, but a try/except/else block. Have a look at [the Python documentation](https://docs.python.org/3/tutorial/errors.html#handling-exceptions).